apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1238,7 +1254,8 @@ NetSimRouterNode.prototype.getNextNodeTowardAddress_ = function (address) {
 
   // Is it another node?
   var destinationNode = _.find(nodes, function (node) {
-    return address === node.getAddress();
+    return address === node.getAddress() ||
+        (node instanceof NetSimRouterNode && address === node.getAutoDnsAddress());
```

Comment:
Yes - right now auto-DNS 'nodes' are actually the router nodes, answering to a second address.